src/network/networkd-sysctl.c

```diff
@@ -28,22 +30,81 @@ static void manager_set_ip_forwarding(Manager *manager, int family) {
                 return; /* keep */
 
         /* First, set the default value. */
-        r = sysctl_write_ip_property_boolean(family, "default", "forwarding", t);
+        r = sysctl_write_ip_property_boolean(family, "default", "forwarding", t, &manager->sysctl_shadow);
         if (r < 0)
                 log_warning_errno(r, "Failed to %s the default %s forwarding: %m",
                                   enable_disable(t), af_to_ipv4_ipv6(family));
 
         /* Then, set the value to all interfaces. */
-        r = sysctl_write_ip_property_boolean(family, "all", "forwarding", t);
+        r = sysctl_write_ip_property_boolean(family, "all", "forwarding", t, &manager->sysctl_shadow);
         if (r < 0)
                 log_warning_errno(r, "Failed to %s %s forwarding for all interfaces: %m",
                                   enable_disable(t), af_to_ipv4_ipv6(family));
 }
 
+static int sysctl_inotify(sd_event_source *source, const struct inotify_event *event, void *userdata) {
+        _cleanup_free_ char *buf = NULL, *path = NULL;
+        Hashmap *sysctl_shadow = ASSERT_PTR(userdata);
```

Comment:
as suggested elsewhere, please make `userdata`actually carry a pointer to a pointer so that this becomes:

```c
Hashmap **sysctl_shadow = ASSERT_PTR(userdata);
```